src/AppBundle/Controller/ArticleInfoController.php

```diff
@@ -98,6 +99,39 @@ public function indexAction(Request $request)
     }
 
     /**
+     * Generate ArticleInfo gadget script for use on-wiki. This automatically points the
+     * script to this installation's API. Pass ?uglify=1 to uglify the code.
+     *
+     * @Route("/articleinfo-gadget.js", name="ArticleInfoGadget")
+     * @link https://www.mediawiki.org/wiki/XTools#ArticleInfo_gadget
+     *
+     * @param Request $request The HTTP request
+     * @return Response
+     */
+    public function gadgetAction(Request $request)
+    {
+        $rendered = $this->renderView('articleInfo/articleinfo.js.twig');
+
+        // SUPER hacky, but it works and is safe.
+        if ($request->query->get('uglify') != '') {
+            $rendered = str_replace('$', '\$', trim($rendered));
+            $rendered = str_replace('"', '\"', trim($rendered));
+            $script = "echo \"$rendered\" | tee temp.js >/dev/null && ";
```

Comment:
This should use a temp file in the sys temp dir, or somewhere. e.g. `$tmpFile = sys_get_temp_dir() . '/xtools_articleinfo_gadget.js';`